aioairq/core.py

```diff
@@ -18,6 +21,53 @@ class DeviceInfo(TypedDict):
     hw_version: str | None
 
 
+class LedTheme(TypedDict):
+    """Container holding the LED themes"""
+    left: str
+    right: str
+
+
+class NightMode(TypedDict):
+    """Container holding night mode configuration"""
+    activated: bool
+    """Whether the night mode is activated"""
```

Comment:
I am nitpicking here, but `""" ... """` is typically used once as a class / method doc-string and not repeated for each field. I suggest to stick to one of the two ways to document it, which is mentioned [here](https://stackoverflow.com/questions/74129732/proper-way-to-document-keys-in-typeddict)

Reply:
On my research I found this link, too. I was not 100% sure if I understood it correctly.

It says:
> **You can't use [type comments](https://peps.python.org/pep-0484/#type-comments)** with the usual `# type: comment` syntax in a `TypedDict` definition. Otherwise the usual two alternatives of putting a docstring above each variable declaration, *or* documenting the members in the class docstring are valid.

I did
> Otherwise the usual two alternatives of putting a docstring above each variable declaration, [...]

Docstring (`""" ... """`) **above** the variable declaration.
I assume (?) `activated: bool` is a variable declaration inside the class's body.

IntelliJ IDE picks up my syntax correctly. (That does not mean it's strictly correct, but in my experience IntelliJ nearly always does thing right).

---

How do you interpret the StackOverflow post? Do you have an example?
I did not want to let the variable go undocumented, because there is a lot of information.

Reply:
Ok, I see that your way is indeed the expected approach, as per [PEP 257](https://peps.python.org/pep-0257/#what-is-a-docstring).
```
1. String literals occurring immediately after a simple assignment at the top level of a module, class, or __init__ method are called “attribute docstrings”.
```
I would only ask you to add, for readability, a new line between an attribute docstring and the next `attribute: type`, cf. [here](https://stackoverflow.com/a/69754746) (although they use single quotes rather incorrectly).

----
I must admit, I interpreted "docstring above the variable declaration" as the comment [here](https://github.com/home-assistant/core/blob/f295382c92b7d3bad7b253f65c37fbb4928c7511/homeassistant/components/energy/data.py#L34), but naturally a comment is no docstring.